--- compat-openssl11.spec	2025-05-21 16:13:26.000000000 +0800
+++ compat-openssl11-patched.spec	2025-10-20 15:22:05.834260600 +0800
@@ -101,6 +101,13 @@
 from the 1.1.1 version and is provided for compatibility with previous
 releases.
 
+%package devel
+Summary: Development package for %{name}
+Requires: %{name} = %{epoch}:%{version}-%{release}
+
+%description devel
+Development package for %{name}.
+
 %prep
 %setup -q -n openssl-%{version}
 
@@ -293,11 +300,18 @@
 rm -f $RPM_BUILD_ROOT/%{_libdir}/engines-1.1/capi.so
 
 # Delete devel files
-rm -rf $RPM_BUILD_ROOT%{_includedir}/openssl
+#rm -rf $RPM_BUILD_ROOT%{_includedir}/openssl
 rm -rf $RPM_BUILD_ROOT%{_mandir}/man3*
 rm -rf $RPM_BUILD_ROOT%{_libdir}/*.so
 rm -rf $RPM_BUILD_ROOT%{_libdir}/pkgconfig
 
+mkdir -p $RPM_BUILD_ROOT{%{_includedir},%{_libdir}}/openssl11
+mv $RPM_BUILD_ROOT%{_includedir}/openssl $RPM_BUILD_ROOT%{_includedir}/openssl11/
+for lib in crypto ssl
+do
+    ln -sf ../lib${lib}.so.%{version} $RPM_BUILD_ROOT%{_libdir}/openssl11/lib${lib}.so
+done
+
 # Install compat config file
 install -m 644 apps/openssl11.cnf $RPM_BUILD_ROOT%{_sysconfdir}/pki/tls/openssl11.cnf
 
@@ -314,6 +328,10 @@
 %dir %{_sysconfdir}/pki/tls
 %attr(0644,root,root) %{_sysconfdir}/pki/tls/openssl11.cnf
 
+%files devel
+%{_libdir}/openssl11
+%{_includedir}/openssl11
+
 %ldconfig_scriptlets
 
 %changelog
